downloads/swan-lake-release-notes/swan-lake-alpha/RELEASE_NOTE.md

```diff
@@ -0,0 +1,1432 @@
+listener Listener ln = new (8080);
+```
+
+##### Improvements to the 'match' Statement
+
+###### Support for More Match Patterns
+
+**List Match Pattern**
+```ballerina
+match v {
+    var     [a, b ] => {
+    // Matches lists with 2 elements.
+    }    var [a, [b, c], d]|var [a, b , c, d] => {
+    // Matches 
+    // - lists with 3 elements where the second element is a list of 2 elements or
+    // - lists with 4 elements
+    }}
+
+```
+
+**Mapping Match Pattern**
+```ballerina
+match v {
+    {a: "hello", b: "world"}     => {
+    // Match mappings that contain the field `a` with
+    // value "hello" and field `b` with value "world".
+    }
+{
+    a:var x} => {
+// Match mappings that contain the field `a`.
+// The value is assigned to the variable `x`
+// and can be assigned within the block.
+io:println(x);
+}
+}
+```
+
+**Error Match Pattern**
+```ballerina
+match v {
+    error ("Message") => {
+    // Match errors with "Message" as the error message.
+    }error () => {
+    // Match all errors.
+    }}
+```
+
+###### Improved Type Narrowing Within Match Statements
+
+When code is executed through each match-clause, the type of the matched expression is narrowed. In the example below, the type of `v` in the last match clause is narrowed to `string`.
+
+```ballerina
+match v {
+    var a if a is int|boolean => {
+        return a.toString();
+    }
+    _ => {
+        return v; // Type of `v` is `string` here.
+    }
+}
+```
+
+##### Support for Cyclic Union Types
+
+Cyclic union types are now supported. A cyclic union type descriptor can directly refer to its identifier in its type descriptor.
+
+```ballerina
+type Integers int|Integers[]|map<Integers>;
+
+Integers intValue = 5;
+Integers intArray = [1, 2, 3, 4];
+Integers intMap = {i: 1, j: 2};
+Integers integers = [intValue, intArray, intMap];
+```
+
+##### Updated Syntax for User-Defined Error Constructions
+
+The error constructor expression now requires the `error` keyword to construct a user-defined error.
+
+**Previous Syntax:**
+```ballerina
+MyError myError = MyError("Message");
+```
+ 
+**New Syntax:**
+```ballerina
+MyError myError = error MyError("Message");
+```
+
+#### Changes on Casting with Errors
+
+Errors cannot be cast away (i.e., if the value that is being cast can be an error, the type to which the cast is attempted should also have a subtype of error). 
+
+The following was allowed previously.
+
+```ballerina
+function foo() returns string|int|error {
+    return error(“Error Message”);
+}
+
+public function main() {
+    string s = <string>foo(); // Cast without considering the error.
+}
+```
+
+This is now disallowed and can be rewritten as follows.
+
+```ballerina
+function foo() returns string|int|error {
+    return error("Error Message");
+}
+
+public function main() {
+    string s = <string>checkpanic foo(); // use`checkpanic` to panic if `foo()` returns an error and then attempt the cast
+}
+```
```

Comment:
Why are we adding this here? This is not in the doc, right?

Reply:
<img width="1387" alt="Screenshot 2021-01-24 at 13 13 23" src="https://user-images.githubusercontent.com/11707273/105624227-09cbb980-5e46-11eb-872c-518a32a2f0d1.png">

Reply:
That is point 2. at L348.

Reply:
Yes, that is the one I am also referring to. Sorry, I can't find anywhere else in which this is duplicated.

Reply:
This seems to be fixed now.